docs/nextflow/reduce-api-calls.md

```diff
@@ -0,0 +1,130 @@
+- 50 concurrent pipeline runs
+- Each run spawns 10,000 tasks
+- Each task pulls a container
+- Total: 500,000 manifest requests
+
+This volume exceeds the 2,000 per minute limit and causes failed tasks and pipeline errors.
+
+## Use Wave freeze to avoid rate limits
+
+Wave freeze builds your container once and stores it in your registry. After the initial build, all container pulls bypass Wave.
+
+**Building without freeze:**
+
+1. Each task requests a manifest from Wave
+1. Wave retrieves the base image from the source registry
+1. Wave builds the image with the fusion layer
+1. Wave returns the modified manifest
+1. Every task creates one API call to Wave
+
+With thousands of concurrent tasks, this approach exceeds rate limits.
+
+**Building with freeze:**
+
+1. Wave builds the image once with your specifications
+1. Wave pushes the complete image to your registry
+1. Wave returns a direct URL to your registry
+1. All future pulls go directly to your registry
+
+With freeze enabled, Wave is removed from the container pull path. Your compute instances pull directly from your registry with no Wave API calls.
+
+## Configure Wave freeze
+
+To configure Wave freeze, add the following configuration to your Nextflow pipeline:
+
+```groovy
+wave.enabled = true
+wave.freeze = true
+wave.build.repository = '<BUILD_REPOSITORY>'
+
+tower.accessToken = '<TOWER_ACCESS_TOKEN>'
+```
+
+Replace the following:
+
+- `<BUILD_REPOSITORY>`: your repository to store and access built containers
+- `<TOWER_ACCESS_TOKEN>`: your Seqera access token
+
+### Container registry selection
+
+**Recommended**: Use Amazon ECR for AWS Batch workloads
+
+Amazon ECR has the following benefits:
+
+- Automatic authentication through IAM roles
+- No manual credential configuration
+- Lowest latency for AWS workloads
+- Simplest setup
+
+**Not recommended**: Private Docker Hub for AWS Batch workloads
+
+Private Docker Hub has the following limitations:
+
+- Requires manual credential configuration on each compute instance
+- Additional security overhead
+- More complex authentication setup
+
+If you use private Docker Hub, configure Docker credentials on each compute instance.
+
+### First pipeline run
+
+When you run your pipeline for the first time with Wave freeze:
+
+1. The Nextflow head job sends your build request to Wave
+1. Wave checks whether the requested images already exist
+1. Wave builds any missing images and pushes them to your registry
+1. Wave returns the final registry URLs
+1. Your compute tasks pull images directly from your registry
+
+The initial build counts against build limits, not pull limits
+
+### Subsequent pipeline runs
+
+When you run your pipeline again:
+
+1. The Nextflow head job contacts Wave to check for existing images
+1. Wave finds the cached images (matched by content hash)
+1. Wave returns the registry URLs immediately without rebuilding
+1. All container pulls go directly to your registry
+1. No Wave API calls occur during task execution
+
```

Comment:
This is partially correct, the pull limit applies also when requesting the freeze container, tho same container will serve many job requests 

https://github.com/seqeralabs/wave/blob/479f50ef9dfd3a630811f6c3dcb0bbe5ad79d280/src/main/groovy/io/seqera/wave/controller/ContainerController.groovy#L266-L267

Reply:
Thanks Paolo, folding it into changes now.